scripts/wildcard-domain-processor/wildcard-expander.ts

```diff
@@ -0,0 +1,263 @@
+/* eslint-disable no-restricted-syntax,no-await-in-loop */
+import path from 'path';
+
+import agtree, {
+    AnyRule,
+    CosmeticRule, DomainListParser,
+    NetworkRule,
+    RuleParser,
+} from '@adguard/agtree';
+
+import { findFilterFiles, readFile, writeFile } from './file-utils';
+import { WildcardDomains } from './wildcard-domains-updater';
+import { DOMAIN_MODIFIERS } from './domain-extractor';
+import { utils } from './utils';
+import { updateContentChecksum } from '../checksum';
+
+/**
+ * Expands wildcards in a network rule AST.
+ * @param ast - The network rule AST to process.
+ * @param wildcardDomains - A map of wildcard domains to their non-wildcard equivalents.
+ * @returns The updated network rule AST with expanded wildcards, or null if no valid domains are left.
+ */
+function expandWildcardsInNetworkRules (
+    ast: NetworkRule,
+    wildcardDomains: WildcardDomains,
+): NetworkRule | null {
+    if (!ast.modifiers) {
+        return ast;
+    }
+
+    const modifiers = ast.modifiers.children;
+    const newPermittedDomains = new Map();
+    const newRestrictedDomains = new Map();
+    let hadWildcard = false;
+    const newModifiers = [];
+
+    for (const modifier of modifiers) {
```

Comment:
```suggestion
    for (const { name, value } of modifiers) {
```

Reply:
do not see the point, since I need modifier bellow